src/main/java/com/example/securitynew/service/impl/UserServiceImpl.java

```diff
@@ -1,40 +1,42 @@
 package com.example.securitynew.service.impl;
 
 import com.example.securitynew.dto.user.UserRegistrationRequestDto;
+import com.example.securitynew.dto.user.UserRequestLoginDto;
 import com.example.securitynew.dto.user.UserResponseDto;
 import com.example.securitynew.exception.RegistrationException;
 import com.example.securitynew.mapper.UserMapper;
-import com.example.securitynew.model.Role;
-import com.example.securitynew.model.Role.RoleName;
 import com.example.securitynew.model.User;
 import com.example.securitynew.repository.UserRepository;
 import com.example.securitynew.service.UserService;
-import java.util.Set;
-import lombok.RequiredArgsConstructor;
-import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import lombok.AllArgsConstructor;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 
 @Service
-@RequiredArgsConstructor
+@AllArgsConstructor
 public class UserServiceImpl implements UserService {
-
-    private BCryptPasswordEncoder bcryptPasswordEncoder;
-    private UserRepository userRepository;
-    private UserMapper userMapper;
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+    private final PasswordEncoder passwordEncoder;
 
     @Override
-    public UserResponseDto register(UserRegistrationRequestDto userRegistrationRequestDto)
+    public UserResponseDto registerUser(UserRegistrationRequestDto userRegistrationRequestDto)
             throws RegistrationException {
         if (userRepository.findByEmail(userRegistrationRequestDto.getEmail()).isPresent()) {
             throw new RegistrationException("Unable to complete registration.");
```

Comment:
Write more informative message, like "user with this email already exist" or smth else